### PR TITLE
Bug fix highlights across periods

### PR DESCRIPTION
**Pull request by ImogenConnorHelleur**
Algorithm adds rule breaks to new period and corrects previous most recent period each time a new period is established.

### CODE REVIEW

**Comment by ImogenConnorHelleur** [comment]
Hi @HorridTom , I think this is ready for you to review and merge if you're okay with the changes. Since we last spoke about this, I've created a new function called `add_rule_breaks_respecting_periods` which avoids highlights across periods by calculating the rule breaks separately for each period and updating the most recent two periods when a new one is added. Resolves issues #4.

Next step will be for me to tidy up the functions.R file and separate it out into separate files.

File: R/functions.R [hunk 1]
```diff
@@ -349,11 +351,78 @@ initialise_limits <- function(data, periodMin,
...
+    #for first period
+    
+    #add rule breaks to all of data
+    limits_table <- add_rule_breaks(x = limits_table)
+    
+  }else if(length(breakpoints) == 1){
+    #for data with 2 periods
+    
+    #split data into sections
+    limits_table_top <- limits_table[1:(counter-1),]
```
[the remark this answers is not in the document]
Comment (ImogenConnorHelleur) [reply R/functions.R:386]: Hi @HorridTom , yes I agree. I'll make these changes next week.

**Comment by ImogenConnorHelleur** [comment]
Hi @HorridTom , I'm really sorry - I had to cut my working day short yesterday so was planning to do some more work tonight. Let me know which ones I can do before tomorrow.

**Comment by HorridTom** [comment]
> Hi @HorridTom , I'm really sorry - I had to cut my working day short yesterday so was planning to do some more work tonight. Let me know which ones I can do before tomorrow.

@ImogenConnorHelleur no problem at all! I think of the above I prefer (1) - and I think that is sufficient... so just renaming the argument from `counter` to `counter_at_period_start` - if you have time to do that before tomorrow am that would be amazing... but again no worries if not!

**Comment by ImogenConnorHelleur** [comment]
Hi @HorridTom, I've made change number 1 and can do 3 (need to check if this is possible) and 4 (not sure where you are meaning to put the comment?) next week

**Comment by HorridTom** [comment]
Great, thanks @ImogenConnorHelleur that's perfect - I honestly think this is enough, but if you prefer to add 3 & 4 too that's fine, we can do that in a new branch. So I'll merge this one first thing tomorrow.

**Pull request merged**